wgpu/src/backend/webgpu.rs

```diff
@@ -1562,6 +1562,40 @@ impl dispatch::InstanceInterface for ContextWebGpu {
         // Devices are automatically polled.
         true
     }
+
+    #[cfg(feature = "wgsl")]
+    fn wgsl_language_features(&self) -> crate::WGSLLanguageFeatures {
+        let mut wgsl_language_features = crate::WGSLLanguageFeatures::empty();
+        if let Some(gpu) = &self.gpu {
+            gpu.wgsl_language_features()
+                .keys()
+                .into_iter()
+                .map(|wlf| wlf.expect("WGSLLanguageFeatures elements should be valid"))
+                .map(|wlf| {
+                    wlf.as_string()
+                        .expect("WGSLLanguageFeatures should be string set")
+                })
+                .filter_map(|wlf| match wlf.as_str() {
+                    "readonly_and_readwrite_storage_textures" => {
+                        Some(crate::WGSLLanguageFeatures::ReadOnlyAndReadWriteStorageTextures)
+                    }
+                    "packed_4x8_integer_dot_product" => {
+                        Some(crate::WGSLLanguageFeatures::Packed4x8IntegerDotProduct)
+                    }
+                    "unrestricted_pointer_parameters" => {
+                        Some(crate::WGSLLanguageFeatures::UnrestrictedPointerParameters)
+                    }
+                    "pointer_composite_access" => {
+                        Some(crate::WGSLLanguageFeatures::PointerCompositeAccess)
+                    }
```

Comment:
issue: Matching on identifiers here duplicates logic that we already have `LanguageExtension::from_ident`.

suggestion: We're also going to need to map from `ImplementedLanguageExtension` to `WGSLLanguageFeatures` anyway when using the `wgpu_core` backend, so let's `match` `LanguageExtension::from_ident(wlf.as_str())` (probably using a separate method on `WGSLLanguageFeatures` or an implementation of `Into`) instead.

Reply:
The problem is that here is webgpu impl, that does not necessary have naga available.

Reply:
Mm, I guess I don't see this as a blocking issue, then. Let's file a follow-up issue to deal with "properly" sharing this, if we can come up with a decent solution.